frontend/packages/kubevirt-plugin/src/utils/resources.ts

```diff
@@ -0,0 +1,4 @@
+import * as _ from 'lodash-es';
+
+export const getLoadedResource = (props, kind) =>
```

Comment:
This kind of functionality should be probably included in the firehose. 
Many components do not care about the loaded prop. They just need the data or undefined/null if the data is not available yet.

- note: previously implemented by WithResources.

Reply:
Components can choose to handle `!loaded` the same as no data, but that should be up to the component. It's a worse API not to have an explicit `loaded` prop. In most cases, you should handle it as well as handle errors.

Reply:
I am not suggesting to make any regression to the API. It is enough Just to enhance it, possibly by adding some flags to the resource (or the whole firehose).

Many components need just the raw data and this would make it easier for them to have less logic or a need to specify getLoadedResource for each resource (like this PR).

Reply:
In most cases, components *should* explicitly handle `loaded`. Changing the API to make it easy to ignore seems like a bad change.

I also think `getLoadedResource` is a not a useful helper because it assumes the property of the data. The prop is not always the kind. It would be better to remove it from the PR.

Reply:
Recently I try to avoid `WithResources` in favour of just `Firehose`. There will be probably a small change to `Firehose` needed when migrating later features, but it's not a recent concern.

The purpose of this `getLoadedResource` function is to provide simple mapping to meet contract with existing component. So far, similar logic was handled via `WithResources` which took care of error handling and potentially passed empty data to the component or tentatively rendered a loader component.

I can improve components in `kubevirt-web-ui-components` to deal with the `Firehose` state machine explicitly. But it's extra complexity as I can't break the "safe-path" for _potentially_  releasing web-ui even in 4.2 timeframe. 

I will try, but definitely I'll definitely need to create new release of `web-ui-components` for that.

Reply:
When looking into the code once again, I think it's better to enhance `Firehose` (see #1711) and we can move checks to children components consistently with the rest of `console`. Please have a look at recent changes in this PR.